SupportBundles: Add bundle encryption based on age (#62501)
grafana/grafana: docs/sources/troubleshooting/support-bundles/index.md

   Grafana downloads the support bundle to an archive (tar.gz) file.

1. Attach the archive (tar.gz) file to a support ticket that you send to Grafana Labs Technical Support.

## Support bundle configuration

You can configure the following settings for support bundles:

```ini
# Enable support bundle creation (default: true)
enabled = true

> # Only server admins can generate and view support bundles (default: true)

Reviewer: nit: could expand on this to say what happens when it's set to false

```suggestion
# Only server admins can generate and view support bundles. When set to false, organization admins can generate and view support bundles (default: true)
```

server_admin_only = true
# If set, bundles will be encrypted with the provided public keys separated by whitespace
public_keys = ""
```

### Support bundle encryption

You can encrypt support bundles with an [age](age-encryption.org) formatted public key.

Ensure [age](https://github.com/FiloSottile/age#installation) is installed on your system.

#### Generating a key pair

```bash
$ age-keygen -o key.txt
Public key: age1ql3z7hjy54pw3hyww5ayyfg7zqgvc7w3j2elw8zmrj2kg5sfn9aqmcac8p
```

> 

Reviewer: nit: I think there could be a new header here to separate key generation, encryption and decryption - something like `#### Encrypting a support bundle`.

To do this, add the public key to the `public_keys` setting in the `support_bundle` section of the Grafana configuration file.

```ini
[support_bundle]
public_keys = "age1ql3z7hjy54pw3hyww5ayyfg7zqgvc7w3j2elw8zmrj2kg5sfn9aqmcac8p"
```

> Multiple public keys can be defined by separating them with whitespace.
> All included public keys will be able to decrypt the support bundle.

Example:

```ini
[support_bundle]
public_keys = "age1ql3z7hjy54pw3hyww5ayyfg7zqgvc7w3j2elw8zmrj2kg5sfn9aqmcac8p ageccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc"
```

New support bundles will be encrypted with the provided public keys and their file extension will be `tar.gz.age`.

#### Decrypting a support bundle

Ensure [age](https://github.com/FiloSottile/age#installation) is installed on your system.

Execute the following command to decrypt the support bundle:

```bash
age --decrypt -i keyfile -o output.tar.gz downloaded.tar.gz.age
```

Example:

```bash

> age --decrypt -i age.key -o data.tar.gz af6684b4-d613-4b31-9fc3-7cb579199bea.tar.gz.age

Author: ```suggestion
age --decrypt -i key.txt -o data.tar.gz af6684b4-d613-4b31-9fc3-7cb579199bea.tar.gz.age
```

public_keys = ""
```

## Encrypting a support bundle

Support bundles can be encrypted with [age](age-encryption.org) before they are sent to
recipients. This is useful when you want to send a support bundle to Grafana through a
channel that is not private.

### Generate a key pair

Ensure [age](https://github.com/FiloSottile/age#installation) is installed on your system.

```bash
$ age-keygen -o key.txt
Public key: age1ql3z7hjy54pw3hyww5ayyfg7zqgvc7w3j2elw8zmrj2kg5sfn9aqmcac8p
```

### Support bundle encryption

Ensure [age](https://github.com/FiloSottile/age#installation) is installed on your system.

Add the public key to the `public_keys` setting in the `support_bundle` section of the Grafana configuration file.

```ini
[support_bundle]
public_keys = "age1ql3z7hjy54pw3hyww5ayyfg7zqgvc7w3j2elw8zmrj2kg5sfn9aqmcac8p"
```

> Multiple public keys can be defined by separating them with whitespace.
> All included public keys will be able to decrypt the support bundle.

Example:

```ini
[support_bundle]
public_keys = "age1ql3z7hjy54pw3hyww5ayyfg7zqgvc7w3j2elw8zmrj2kg5sfn9aqmcac8p age1yu8vzu554pv3klw46yhdv4raz36k5w3vy30lpxn46923lqngudyqvxacer"
```

> Restart Grafana and new support bundles will be encrypted with the provided

Reviewer 2: ```suggestion
When you restart Grafana, new support bundles will be encrypted with the provided
```

> public keys and their file extension will be `tar.gz.age`.

Reviewer 2: ```suggestion
public keys. The support bundle file extension is `tar.gz.age`.
```
